src/components/ProgramForm/ThreeColProgram/List/Recommender/RecommenderTopicSelector.tsx

```diff
@@ -0,0 +1,168 @@
+import PACKAGES from '@/consts/packages';
+import { useState, useCallback } from 'react';
+import { Button, Tooltip } from 'antd';
+import {
+  FundOutlined,
+  FileTextOutlined,
+  FilePdfOutlined,
+  FileImageOutlined,
+  VideoCameraAddOutlined,
+  AudioOutlined,
+  YoutubeOutlined,
+  InteractionOutlined,
+  CarryOutOutlined,
+  PercentageOutlined,
+  PlusOutlined,
+  MinusOutlined,
+} from '@ant-design/icons';
+
+import '../types.css';
+import { TopicType } from '@/services/escola-lms/enums';
+import { FormattedMessage, useModel } from 'umi';
+import { topicTypeToSettingName } from '@/pages/Settings/global';
+
+export const RecommenderTopicSelector: React.FC<{
+  onSelected: (type: TopicType) => void;
+  positionsToHide?: (TopicType | 'lesson')[];
+}> = ({ onSelected, positionsToHide }) => {
+  const [open, setOpen] = useState<boolean>(false);
+
+  const { initialState } = useModel('@@initialState');
+
+  const setSelected = useCallback((type: TopicType) => {
+    setOpen(false);
+    onSelected(type);
+  }, []);
+
+  const topicTypeIsDisabled = useCallback(
+    (type: TopicType) => {
+      const key = Object.keys(TopicType)
+        .filter((x) => TopicType[x] == type)
+        .pop();
+
+      const value =
+        key &&
+        initialState?.config?.filter((item) => item.key === topicTypeToSettingName(key))[0]?.data;
+
+      if (!!value) {
+        return true;
+      }
+
+      switch (type) {
+        case TopicType.GiftQuiz:
+          return !(initialState?.packages && initialState?.packages?.[PACKAGES.TopicTypeGift]);
+        case TopicType.SCORM:
+          return !(initialState?.packages && initialState?.packages?.[PACKAGES.Scorm]);
+        case TopicType.Project:
+          return !(initialState?.packages && initialState?.packages?.[PACKAGES.TopicTypeProject]);
+        default:
+          return !(initialState?.packages && initialState?.packages?.[PACKAGES.TopicTypes]);
+      }
+    },
+    [initialState],
+  );
+
+  return (
+    <div
+      className={`topic-types-selector topic-types-selector__recommender-wrapper
+       ${open ? 'topic-types-selector--visible' : ''}`}
+    >
+      <Button
+        type="text"
+        size="small"
+        onClick={() => setOpen((prev) => !prev)}
+        className="topic-types-selector__recommender--add-btn"
+        icon={open ? <MinusOutlined /> : <PlusOutlined />}
+      >
+        <FormattedMessage id="recommender.add_new_exercise" />
+      </Button>
+
+      <div
+        className={`${'topic-types-selector__recommender'}
+          ${open ? 'topic-types-selector__recommender--active' : ''}`}
+      >
+        {!positionsToHide?.includes(TopicType.RichText) &&
+          !topicTypeIsDisabled(TopicType.RichText) && (
```

Comment:
I see room for improvements here. I would probably move it to custom fn e.g.

```ts
const isTopicVisible = (type: TopicType): boolean => !positionsToHide?.includes(type) && !topicTypeIsDisabled(type)
```

Also I would consider mapping the topic types here, as You are repeating a lot of code.